src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -186,6 +186,11 @@ private IEnumerable<TypeInferenceInfo> InferTypeInArrowExpressionClause(ArrowExp
                     return InferTypeInBaseMethodDeclaration(arrowClause.Parent as BaseMethodDeclarationSyntax);
                 }
 
+                if (arrowClause.IsParentKind(SyntaxKind.GetAccessorDeclaration))
+                {
+                    return InferTypeInGetAccessorDeclaration(arrowClause.Parent as AccessorDeclarationSyntax);
+                }
+
```

Comment:
We should also handle "SetAccessors" (and probably add/remove as well).  In these cases, the inferred type would be 'void'.  You should be able to validate that by doing something like this:   ```class C { int P { set => Foo(value); } }``` and generating a method for 'Foo'.  We'd want it to infer 'void' as the return type.

Reply:
I now added some tests in `GenerateMethodTests.cs` as well for both the getter and the setter situation. It indeed worked as expected after applying pattern matching.